examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,273 @@
+        *TARGET_DIM, interpolation=interpolation
+    )(inputs)
+
+    # First conv block without Batch Norm.
+    x = layers.Conv2D(filters=filters, kernel_size=7, strides=1, padding="same")(inputs)
+    x = layers.LeakyReLU(ALPHA)(x)
+
+    # Second conv block with Batch Norm.
+    x = layers.Conv2D(filters=filters, kernel_size=1, strides=1, padding="same")(x)
+    x = layers.LeakyReLU(ALPHA)(x)
+    x = layers.BatchNormalization()(x)
+
+    # Intermediate resizing as bottleneck.
+    bottleneck = layers.experimental.preprocessing.Resizing(
+        *TARGET_DIM, interpolation=interpolation
+    )(x)
+
+    # Residual passes.
+    for _ in range(num_res_blocks):
+        x = res_block(bottleneck)
+
+    # Projection.
+    x = layers.Conv2D(
+        filters=filters, kernel_size=3, strides=1, padding="same", use_bias=False
+    )(x)
+    x = layers.BatchNormalization()(x)
+
+    # Skip connection
+    x = layers.Add()([bottleneck, x])
+
+    # Final resized image
+    x = layers.Conv2D(filters=3, kernel_size=7, strides=1, padding="same")(x)
+    final_resize = layers.Add()([naive_resize, x])
+
+    return final_resize
+
+
+"""
+## Visualize the outputs of the learnable resizing module
+
+Here, we visualize how the resized images would look like after being passed through the
+random weights of the resizer. 
+"""
+
+sample_images, _ = next(iter(train_ds))
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learnable_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+## Model building utility
+"""
+
+
+def get_model():
+    backbone = tf.keras.applications.DenseNet121(
+        weights=None, include_top=True, classes=2
+    )
+    backbone.trainable = True
+
+    inputs = layers.Input((INP_DIM[0], INP_DIM[1], 3))
+    x = layers.experimental.preprocessing.Rescaling(scale=1.0 / 255)(inputs)
+    x = learnable_resizer(x)
+    outputs = backbone(x)
+
+    return tf.keras.Model(inputs, outputs)
+
+
+"""
+## Compile and train our model with learnable resizer
+"""
+
+model = get_model()
+model.compile(
+    loss=keras.losses.CategoricalCrossentropy(label_smoothing=0.1),
+    optimizer="sgd",
+    metrics=["accuracy"],
+)
+model.fit(train_ds, validation_data=validation_ds, epochs=EPOCHS)
+
+"""
+## Visualize the outputs of the trained visualizer
+"""
+
+learned_resizer = tf.keras.Model(model.input, model.layers[-2].output)
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learned_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+As we can see the visuals of the images have improved with training. Additionally, you
```

Comment:
Tip: for accessibility, I'd recommend avoiding using "as we can see":

```suggestion
The plots show that the images have improved with training. Additionally, you
```